summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/ScaldingPlatform.scala

```diff
@@ -652,67 +652,43 @@ class Scalding(
   def plan[T](prod: TailProducer[Scalding, T]): PipeFactory[T] =
     Scalding.plan(options, prod)
 
-  protected def ioSerializations: List[Class[_ <: HSerialization[_]]] = List(
-    classOf[org.apache.hadoop.io.serializer.WritableSerialization],
-    classOf[cascading.tuple.hadoop.TupleSerialization],
-    classOf[com.twitter.chill.hadoop.KryoSerialization]
-  )
-
   def withRegistrars(newRegs: List[IKryoRegistrar]) =
     new Scalding(jobName, options, transformConfig, newRegs ++ passedRegistrars)
 
-  def withConfigUpdater(fn: SummingbirdConfig => SummingbirdConfig) =
+  def withConfigUpdater(fn: Config => Config) =
     new Scalding(jobName, options, transformConfig.andThen(fn), passedRegistrars)
 
-  def updateConfig(conf: Configuration) {
-    val scaldingConfig = SBChillRegistrar(ScaldingConfig(conf), passedRegistrars)
-    Scalding.logger.debug("Serialization config changes:")
-    Scalding.logger.debug("Removes: {}", scaldingConfig.removes)
-    Scalding.logger.debug("Updates: {}", scaldingConfig.updates)
+  def configProvider(hConf: Configuration): Config = {
+    import com.twitter.scalding._
+    import com.twitter.chill.config.ScalaMapConfig
+    val conf = Config.hadoopWithDefaults(hConf)
 
-    // now let the user do her thing:
-    val transformedConfig = transformConfig(scaldingConfig)
+    val kryoReg = new IterableRegistrar(passedRegistrars)
 
-    Scalding.logger.debug("User+Serialization config changes:")
-    Scalding.logger.debug("Removes: {}", transformedConfig.removes)
-    Scalding.logger.debug("Updates: {}", transformedConfig.updates)
+    val initKryo = conf.getKryo match {
+      case None =>
+        new serialization.KryoHadoop(ScalaMapConfig(conf.toMap))
+      case Some(kryo) => kryo
+    }
+    conf
+      .setSerialization(
+        Left((classOf[serialization.KryoHadoop], initKryo.withRegistrar(kryoReg))), Nil)
```

Comment:
can we use the `.serSerialization(Right(...))` case if passedRegistrars.isEmpty?